src/middleware.ts

```diff
@@ -368,22 +368,23 @@ export const persist =
             )
           } else {
             stateFromStorageInSync = deserializedStorageValue.state
-            set(deserializedStorageValue.state)
           }
         }
       })
       .then((migratedState) => {
         if (migratedState) {
           stateFromStorageInSync = migratedState as S
-          set(migratedState as PartialState<S, keyof S>)
-          return setItem()
         }
-      })
-      .then(() => {
+
         stateFromStorageInSync = merge(
           stateFromStorageInSync as S,
           configResult
         )
+
+        set(stateFromStorageInSync as S)
```

Comment:
```suggestion
        set(stateFromStorageInSync as S, true)
```
I feel like we should replace it, but i'm not so sure.
btw, can we call `set` in sync?

Reply:
hm, it was already called in the previous code too.

Reply:
the set in sync doesn't seem to trigger a render if that's what you're worried about

Reply:
if it doesn't error, it's fine. I mistook it was only for async.